bluetti_mqtt/core/devices/eb3a.py

```diff
@@ -50,7 +50,7 @@ def polling_commands(self) -> List[QueryRangeCommand]:
         return [
             QueryRangeCommand(0x00, 0x0A, 0x28),
             QueryRangeCommand(0x00, 0x46, 0x15),
-            QueryRangeCommand(0x0B, 0xB9, 0x1C)
+            QueryRangeCommand(0x0B, 0xDA, 1)
```

Comment:
From trial and error it looks like this range is enough to get all currently know fields. 

Is there a write-up how querying works? Just guessing from the code makes me feel a bit lost especially since the first and the last query below both contain ac_output_on and dc_output_on even tough the ranges don't seem to overlap.
 
`
# QueryRangeCommand(0x00,0x0A,0x28),
{
    "device_type": "EB3A",
    "serial_number": XXXXXXXXXXXXXXX,
    "arm_version": Decimal("2053.07"),
    "dsp_version": Decimal("2052.04"),
    "dc_input_power": 0,
    "ac_input_power": 0,
    "ac_output_power": 0,
    "dc_output_power": 4,
    "power_generation": Decimal("0"),
    "total_battery_percent": 100,
    "ac_output_on": False,
    "dc_output_on": True
}

# QueryRangeCommand(0x00,0x46,0x15),
{
    "ac_input_voltage": Decimal("224.7"),
    "dc_input_voltage": Decimal("0")
}

# QueryRangeCommand(0x0B,0xDA,1)
{
    "led_mode": <LedMode.OFF: 4>
}

# QueryRangeCommand(0x0B,0xB9,0x1C)
{
    "ac_output_on": False,
    "dc_output_on": True
}
`

Reply:
The first argument is what I've been calling the "page". The 0x00 page is read-only, and has a bunch of data. The 0x0B page has various settings, and their current state. I wrote a program to scan for other pages a while back, and there are others (0x13 has wifi configs for devices that support it), but all the interesting data is on those two pages. The second and third arguments are the start offset and the length of the range query in "offsets". At each offset is one 16bit field, so the byte length is twice the range query length.

When you look at a specific device and see `self.struct.add_bool_field('ac_output_on', 0x0B, 0xBF)`, that's basically saying that the settable ac output switch is on page 0x0B at offset 0xBF, and is stored as a 16-bit integer value (0 is False and 1 is True).

The reason that the first and last query both contain `ac_output_on` and `dc_output_on` is because the 0x00 page has a read-only copy of that data, and the 0x0B page is where it can be changed. My guess is that this makes it easier in the app to just poll page 0x00, offset 0x24-0x27 to display on the main screen of the app, but that's just a guess.

Reply:
Thanks for the explanation, it really helps. It might be worth replicating it to a readme file or wiki for future reference.